src/components/course/data/hooks.js

```diff
@@ -168,37 +187,54 @@ export function useCoursePriceForUserSubsidy(activeCourseRun, userSubsidy) {
         return null;
       }
 
-      const priceDetails = {
-        list: numberWithPrecision(listPrice),
+      const onlyListPrice = {
+        list: listPrice,
       };
-      if (!userSubsidy) {
-        return priceDetails;
-      }
 
-      const { discountType, discountValue } = userSubsidy;
-      let discountedPrice;
+      if (userSubsidyApplicableToCourse) {
+        const { discountType, discountValue } = userSubsidyApplicableToCourse;
+        let discountedPrice;
 
-      if (discountType === SUBSIDY_DISCOUNT_TYPE_MAP.PERCENTAGE) {
-        discountedPrice = listPrice - (listPrice * (discountValue / 100));
-      }
+        if (discountType === SUBSIDY_DISCOUNT_TYPE_MAP.PERCENTAGE) {
+          discountedPrice = listPrice - (listPrice * (discountValue / 100));
+        }
 
-      if (discountType === SUBSIDY_DISCOUNT_TYPE_MAP.ABSOLUTE) {
-        discountedPrice = Math.max(listPrice - discountValue, 0);
-      }
+        if (discountType === SUBSIDY_DISCOUNT_TYPE_MAP.ABSOLUTE) {
+          discountedPrice = Math.max(listPrice - discountValue, 0);
+        }
 
-      if (isDefinedAndNotNull(discountedPrice)) {
-        priceDetails.discounted = numberWithPrecision(discountedPrice);
-      } else {
-        priceDetails.discounted = priceDetails.list;
+        if (isDefinedAndNotNull(discountedPrice)) {
+          return {
+            ...onlyListPrice,
+            discounted: discountedPrice,
+          };
+        }
+        return {
+          ...onlyListPrice,
+          discounted: onlyListPrice.list,
+        };
       }
 
-      return priceDetails;
+      // Case 2: No subsidy, no offers for course
```

Comment:
tiniest of nits: Maybe `Case 2: No subsidies available for course` as offers are also subsidies here 😄